examples/diet.jl

```diff
@@ -71,16 +72,30 @@ function SolveDiet()
 
     # Solve
     println("Solving original problem...")
-    status = solve(m)
-    PrintSolution(status, foods, buy)
+    JuMP.optimize!(m)
+    term_status = JuMP.termination_status(m)
+    primal_status = JuMP.primal_status(m)
+
+    if term_status == MOI.Success && primal_status == MOI.FeasiblePoint
+        PrintSolution(:Optimal, foods, buy)
```

Comment:
Using `:Optimal` is weird and no longer good style. This code should be restructured to more naturally for how MOI statuses work. Maybe
```julia
if primal_status == MOI.FeasiblePoint
    print_solution(foods, buy)
else
    println("No solution.")
end
if term_status != MOI.Success
    println("The solver finished with non-success status $term_status.")
end
```

Reply:
Thanks!

Reply:
> Can term_status == MOI.Success also mean suboptimal solution?

http://www.juliaopt.org/MathOptInterface.jl/stable/apireference.html#MathOptInterface.TerminationStatusCode

Depends on the algorithm and what you mean by suboptimal. If suboptimal means not globally optimal, definitely. Gradient descent finish successfully when the norm of the gradient is sufficiently small, which doesn't in general say anything about global optimality. If you're using a branch-and-bound MIP solver, then yes, you'd expect that `Success` means global optimality.